### PR TITLE
Add the chip component

### PR DESCRIPTION
**Pull request by anthonydillon**
## Done
Add the chip component

Fixes https://github.com/canonical-web-and-design/vanilla-framework/issues/3143

## QA
- Pull code
- Run `./run`
- Open http://0.0.0.0:8101/docs/patterns/chip
- Check the content is ok
- Open http://0.0.0.0:8101/docs/examples/patterns/chip
- Check the component renders as expected

### CODE REVIEW

**Comment by lyubomir-popov** [comment]
@anthonydillon The padding is a bit off on the chip compared to my branch:
A couple of questions:
image

- Why the line-height?
- Why no padding-top:.2rem?

These two in combination make the text not centered properly compared to what I experimented with:

image

**Comment by anthonydillon** [comment]
@lyubomir-popov thanks, that was due to edits to fix the dismiss button into the chip. Updated now.

File: templates/docs/patterns/chip.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+```scss
+@import 'patterns_chip';
+@include vf-p-chip;
+```
+
+For more information see [Customising Vanilla](/docs/customising-vanilla/) in your projects, which includes overrides and importing instructions.
+
+### Design
+
+Unfortunately there are no design specs available for this component.
```
Comment (lyubomir-popov) [inline templates/docs/patterns/chip.md:30]: Noted, I'll add design spec ASAP. Issue here: https://github.com/canonical-web-and-design/vanilla-framework/issues/3161

Comment (anthonydillon) [reply templates/docs/patterns/chip.md:30]: Thanks, I didn't know if you wanted to do them anymore.

**Comment by bartaz** [comment]
In IE text of Dismiss button is partially visible:

Screenshot 2020-07-09 at 10 13 57

**Comment by bartaz** [comment]
While the existing minimal example is fine for embedding in the documentation, I think it would be worth having a bigger example as well, that would cover more cases: chip with dismiss, chip without dismiss, chip with very long label, many chips and how they wrap...

**Comment by lyubomir-popov** [comment]
Forgot to mention yesterday, but the `Owner:` string shouldbe uppercase and smaller font as in the zeplin design:
https://app.zeplin.io/project/5dd5644794f8de4540d34860/dashboard?seid=5eddf432ba59c2b364418979

**Comment by anthonydillon** [comment]
> In IE text of Dismiss button is partially visible:
> 
> Screenshot 2020-07-09 at 10 13 57

This is an issue with `%vf-hide-text`. IE does not support transitioning values set with `calc()`  therefore a solution could be to switch to using `2vw`. I will tackle that as [a separate issue](https://github.com/canonical-web-and-design/vanilla-framework/issues/3162) so its easier to discover and revert if needed in the future.

File: templates/docs/patterns/chip.md (hunk 1 above)
Comment (bartaz) [reply templates/docs/patterns/chip.md:30]: @anthonydillon 

We talked with @lyubomir-popov and @spencerbygraves if there is any need to maintain the design (in a repo) at all.
The design team is creating the specs in Zeplin, so they are available for devs during the component build, and the design spec repo seems to be unmaintained and not really used by anyone.

So we thought that probably it's not worth maintaining it, and adding new designs there.

In this case we would just remove the `Design` section from this docs page completely, and later we could archive the design spec repo and maybe remove the Design section from other docs pages as well.

Any thoughts?

**Comment by anthonydillon** [comment]
> Forgot to mention yesterday, but the `Owner:` string shouldbe uppercase and smaller font as in the zeplin design:
> https://app.zeplin.io/project/5dd5644794f8de4540d34860/dashboard?seid=5eddf432ba59c2b364418979

Done.

**Comment by anthonydillon** [comment]
> While the existing minimal example is fine for embedding in the documentation, I think it would be worth having a bigger example as well, that would cover more cases: chip with dismiss, chip without dismiss, chip with very long label, many chips and how they wrap...

Sure I added:
- long-values
- multiple
- with-dismiss
- without-dismiss

File: templates/docs/patterns/chip.md (hunk 1 above)
Comment (anthonydillon) [reply templates/docs/patterns/chip.md:30]: Ok if that is the way in which the design team want to continue to drive the framework that's fine by me. I think we can archive the repo now and start removing the design sections soon.

**Comment by lyubomir-popov** [comment]
@anthonydillon can you please retrigger the build?

I don't see the chips on the examples page 
image
and there seems to be a linting error:
image

**Comment by bartaz** [comment]
Looks like example HTML files are missing from the commits.

And why was the old demo triggered for it instead of new one?

**Comment by anthonydillon** [comment]
Sorry of course, forgot to add. They are there now.

**Comment by lyubomir-popov** [comment]
@anthonydillon the chips are misaligned, I believe it is due to inline flex:
image

**Comment by lyubomir-popov** [comment]
Also the distance between the chips is very small with margin-bottom:0 compared to my prototype:

image
image


Is this to fit them in the search field? And what markup is needed to ensure outside of a search field they look as in the first picture?

**Comment by lyubomir-popov** [comment]
@anthonydillon based a pr on this, more explanations on it: https://github.com/canonical-web-and-design/vanilla-framework/pull/3179

**Comment by anthonydillon** [comment]
@lyubomir-popov thanks so should this one be closed?

**Comment by lyubomir-popov** [comment]
@anthonydillon maybe review the other one first, and if you agree with the proposed changes then close this one?

**Pull request closed without merging**